Documentation book (#263)
status-im/nim-json-rpc: docs/src/receiving_requests.md

# Receiving a JSON-RPC request

Before receiving any request, you should have already [established a connection](./connecting.md).

> When a request is received, the router matches it to a server method that was previously registered with a matching name. If no matching server method can be found the request is dropped, and an error is returned to the client if the client requested a response.

Reviewer: ```suggestion
When a request is received, the router matches it to a server method by name. If no matching server method can be found the request is dropped, and an error is returned to the client if the client requested a response.
```

When an RPC-invoked server method throws an exception, the server will handle the exception and (when applicable) send an error response to the client with a description of the failure.

JSON-RPC is an inherently asynchronous protocol. Multiple concurrent requests are allowed. Methods are invoked as the requests are processed, even while prior requests are still running.

## Registering methods

> The `rpc` macro accepts a list of proc definitions which are turned into async procedures and registered as RPC methods. Procedure overload is not supported. A format flavor supporting the parameters and return type must be set. The `RpcConv` defined in the [flavors section](./format_conversion.md) is used in the following example:

Reviewer: the concept of "format flavors" needs to be introduced before it's used here
